recipes/cpr/all/conanfile.py

```diff
@@ -138,9 +138,16 @@ def validate(self):
         if is_msvc(self) and self.options.shared and "MT" in self.settings.compiler.runtime:
             raise ConanInvalidConfiguration("Visual Studio build for shared library with MT runtime is not supported")
 
-        if tools.Version(self.version) == "1.9.0" and self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "6":
+        if tools.Version(self.version) >= "1.9.0" and self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "6":
             raise ConanInvalidConfiguration("{}/{} doesn't support gcc < 6".format(self.name, self.version))
 
+        if self.settings.compiler.get_safe("cppstd"):
+            # see https://github.com/libcpr/cpr/issues/796#issuecomment-1195056626
+            if tools.Version(self.version) <= "1.9.0":
+                tools.check_min_cppstd(self, 11)
+            else:
+                tools.check_min_cppstd(self, 17)
```

Comment:
This requires extra logic to make sure a higher compiler is picked

conan-io/conan#8002 is the old way... here's an updated recipe https://github.com/conan-io/conan-center-index/blob/01b909dfbcb5de24fd411444cecd05a7637c06c8/recipes/simdjson/all/conanfile.py#L51